crates/oxc_prettier/src/lib.rs

```diff
@@ -23,6 +23,19 @@ use crate::{doc::Doc, format::Format, printer::Printer};
 
 pub use crate::options::{ArrowParens, EndOfLine, PrettierOptions, QuoteProps, TrailingComma};
 
+type GroupId = u32;
+#[derive(Default)]
+struct GroupIdBuilder {
```

Comment:
Let's use the newtype idiom here https://doc.rust-lang.org/rust-by-example/generics/new_types.html

Just call it `GroupId` and expose the API `next_id`.

Reply:
Done!